events/api.py

```diff
@@ -1840,16 +1841,48 @@ class RegistrationSerializer(LinkedEventsSerializer, RegistrationBaseSerializer)
         queryset=Event.objects.all(),
     )
 
+    only_admin_visible_fields = ("created_by", "last_modified_by", "registration_users")
+
+    def _create_registration_users(self, registration, registration_users):
+        for registration_user in registration_users:
+            user_instance, created = RegistrationUser.objects.get_or_create(
```

Comment:
Is it possible that existing RegistrationUsers' emails are updated? In that case `update_or_create` would be more suitable, and perhaps there could be another private method `_update_registration_users` for those RegistrationUsers that already exist in the DB?

Reply:
Though the current logic in these cases recreates the users with new emails and in a way already "updates" them. Updating the usual way as suggested in my previous comment should be a bit more efficient than deleting and the recreating.

Reply:
My Friday-brains didn't find perfect solution to implement what you suggested, but I changed
```
user_instance, created = RegistrationUser.objects.get_or_create(
                registration=registration, **registration_user
)
```
to
```
user_instance, created = RegistrationUser.objects.update_or_create(
                email=registration_user["email"],
                registration=registration,
                defaults=registration_user,
            )
```
so user can update `language` value via API.

Reply:
You would need to do it like this to implement what I suggested:
```
RegistrationUser.objects.update_or_create(id=<id>, registration=registration, defaults={"email": <email>, "language": <language>})
```
where `id` can be `None` for a new user.

Reply:
I am thinking that for robust data integrity the `unique_together` should probably be kept, but in the case where there would be another registration user with the same email address as in the `update` action data, somebody (API client, admin, etc.) would need to delete the existing entry based on the validation error and recreate it with the updated email. So perhaps just return the validation error in this case and let the API user handle the error?

Reply:
Good point. Updating registration users so that it will cause integrity errors is kind of edge case. Removed the migration file to remove `unique_together`. Fixed in https://github.com/City-of-Helsinki/linkedevents/pull/631/commits/025bfe395b5fe917bdc02ca0a4d29814470c29bd